### PR TITLE
Master new pages

### PR DESCRIPTION
**Pull request by reave006**
My first Pull Request!  Please let me know if I'm missing any required information.  This pull request includes a basic person class (which will probably be replaced by Allans user class) modifications to the createaccount page, creation of the about page as well as the modification of the dashboard to include the required fields to use MoviesDB.

### CODE REVIEW

**Comment by joeyfreeman** [comment]
The password validation is pretty tricky to pass. Test123@ should be valid (based on the code comments) but I'm still receiving "Incorrect Password Format". We should tell the user what the requirements are.
The Account Details window opens correctly, but we already have a page on the dashboard for that.
For the About page, the missing full names are:
Maria Milena Arcena
Diana Anguiano
Robert Carleton

File: PlotTalk/PlotTalk.pro [hunk 1]
```diff
@@ -68,14 +78,14 @@ test {
         Test/testtvshow.h \
         Test/testjsonconnection.h \
         Test/testdatabasemanager.h \
-		Test/testreply.h
+                Test/testreply.h
```
Comment (allanhaywood) [inline PlotTalk/PlotTalk.pro]: Fix indentation.

File: PlotTalk/PlotTalk.pro [hunk 2]
```diff
@@ -68,14 +78,14 @@ test {
...
 
     SOURCES += \
         Test/testmain.cpp \
         Test/testtvshow.cpp \
         Test/testjsonconnection.cpp \
         Test/testdatabasemanager.cpp \
-		Test/testreply.cpp
+                Test/testreply.cpp
```
Comment (allanhaywood) [inline PlotTalk/PlotTalk.pro]: Fix indentation.

**Review by allanhaywood — CHANGES_REQUESTED** [review]
Please add unit tests for Account manager, especially things like validating negative and positive input for username/email/password etc.

NOTE: Don't worry about adding test for code that will be removed when you switch over to using the dbmanager.

**Comment by allanhaywood** [comment]
NOTE: This is quite a big review. In the future, try and keep the reviews smaller, something that still contains a functional set of changes, but isn't so large to make the review cumbersome.

For example, one review for the new pages being added, and another review for the account manager functionality.

**Comment by reave006** [comment]
The names were changed to full names, the password field has been modified to ensure that good passwords actually work (don't know why the ones I choose worked and the ones you all choose didn't--probably ghosts).   Had to move password checking into account manager and reorganize some stuff to make unit tests work, but it was done and they are added.  All tests pass.

File: PlotTalk/PlotTalk.pro [hunk 3]
```diff
@@ -36,20 +40,28 @@ SOURCES += \
...
+    Src/person.cpp \
+    Src/about.cpp \
     Src/jsonconnection.cpp \
     Src/databasemanager.cpp \
     Src/reaction.cpp \
     Src/reply.cpp \
     Src/review.cpp \
     Src/comment.cpp
-           
+
```
Comment (allanhaywood) [inline PlotTalk/PlotTalk.pro:54]: Remove extra lines.

File: PlotTalk/PlotTalk.pro [hunk 4]
```diff
@@ -68,14 +80,16 @@ test {
         Test/testtvshow.h \
         Test/testjsonconnection.h \
         Test/testdatabasemanager.h \
-		Test/testreply.h
+        Test/testaccountmanager.h \
+                Test/testreply.h
```
Comment (allanhaywood) [inline PlotTalk/PlotTalk.pro]: Fix indentation.

File: PlotTalk/PlotTalk.pro [hunk 5]
```diff
@@ -68,14 +80,16 @@ test {
...
 
     SOURCES += \
         Test/testmain.cpp \
         Test/testtvshow.cpp \
         Test/testjsonconnection.cpp \
         Test/testdatabasemanager.cpp \
-		Test/testreply.cpp
+        Test/testaccountmanager.cpp \
+                Test/testreply.cpp
```
Comment (allanhaywood) [inline PlotTalk/PlotTalk.pro]: Fix indentation.

File: PlotTalk/Src/accountmanager.cpp [hunk 6]
```diff
@@ -27,107 +30,124 @@ AccountManager::~AccountManager()//destructor
 }
 
 
-
-
-bool AccountManager::createAccount(QString &first, QString &last, QString &email, QString &handle, QString &password)
+bool AccountManager::createAccount(QString &first, QString &last, QString &Email, QString &handle, QString &password)
```
Comment (allanhaywood) [inline PlotTalk/Src/accountmanager.cpp:33]: Is there a reason these are all being passed in as references?
If there isn't, please change them to pass by value.

File: PlotTalk/Src/accountmanager.cpp [hunk 7]
```diff
@@ -27,107 +30,124 @@ AccountManager::~AccountManager()//destructor
...
             return true;//returns true that the account was created
         }
-        file.close();//closes the file without writing, something went wrong
-    }
 
     return false;//returns false, something went wrong
 }
 
-bool AccountManager::findPersonByHandle(Person &PassBack, QString &handleToCheck)
+bool AccountManager::findPersonByHandle(QString &handleToCheck)
```
Comment (allanhaywood) [inline PlotTalk/Src/accountmanager.cpp:54]: Is there a reason this is being passed in as references?
If there isn't, please change them to pass by value.

File: PlotTalk/Src/accountmanager.cpp [hunk 8]
```diff
@@ -27,107 +30,124 @@ AccountManager::~AccountManager()//destructor
...
-    in>>email;
-    in>>first;
-    in>>last;
-    in>>handle;
-    in>>password;//stores the variables from the file into strings
+        return false;//person is not in list
+    }
 
-        if (handle==handleToCheck)//once stored, checks the handle to see if it's a match
+    for (int i =0; i<peopleList.size();i++)
```
Comment (allanhaywood) [inline PlotTalk/Src/accountmanager.cpp:62]: Fix spacing on i =0

File: PlotTalk/Src/accountmanager.cpp [hunk 9]
```diff
@@ -27,107 +30,124 @@ AccountManager::~AccountManager()//destructor
...
 }
 
 Person AccountManager::getCurrentAccount()
 {//retuns the account information of the account held in the program
-return ThisGuy;//useful for getting info into various pages without searching the database
+return thisGuy;//useful for getting info into various pages without searching the database
 }
 
-bool AccountManager::findPersonByEmail(Person &PassBack, QString &emailtoCheck)
+bool AccountManager::findPersonByEmail(QString &emailtoCheck)
```
Comment (allanhaywood) [inline PlotTalk/Src/accountmanager.cpp:78]: Same comment about pass by value.

File: PlotTalk/Src/accountmanager.cpp [hunk 10]
```diff
@@ -27,107 +30,124 @@ AccountManager::~AccountManager()//destructor
...
-           PassBack.handle=handle;
-           PassBack.password=handle;//if it's a match gathers the information into passback to reutrn to the client
+            return true;//person is in list
+        }
+    }
+
+    return false;//list is not empty but person is not in list
+}
+
+int AccountManager::checkFields(QString &fName, QString &lName, QString &handle, QString &email, QString &password)
```
Comment (allanhaywood) [inline PlotTalk/Src/accountmanager.cpp:97]: Same comment about pass by value.

Pass by reference is only needed if the function/method will be modifying the values passed in, and you want the caller to see those changes, or it is a very large in memory object, and you want to avoid copying it.

File: PlotTalk/Src/accountmanager.cpp [hunk 11]
```diff
@@ -27,107 +30,124 @@ AccountManager::~AccountManager()//destructor
...
+/*
 
-           ThisGuy=PassBack;//stores the new user as the current in the program
+Password filter that matches the NSA Password filter DLL ENPASFILT.DLL. At least 1 small-case letter At least 1 Capital letter At least 1 digit At least 1 special character Length should be between 8-30 characters. Spaces allowed The sequence of the characters is not important.
 
-           file.close();
-           return true;
+*/
+    int num=0;
+    if(validEmail->validate(email,num)==2)//valid email
```
Comment (allanhaywood) [inline PlotTalk/Src/accountmanager.cpp:111]: A lot of nesting going on here, please simplify.

File: PlotTalk/Src/accountmanager.h [hunk 12]
```diff
@@ -5,37 +5,29 @@
...
-    QString fileName="Data.txt";
-    Person ThisGuy;
+    QList<Person> peopleList;//will be replaced by database manager
+    Person thisGuy;
 
 public:
 
     static AccountManager* getInstance();
-    bool createAccount(QString &first, QString &last, QString &email, QString &handle, QString &password);//done
+    bool createAccount(QString &first, QString &last, QString &Email, QString &handle, QString &password);//done
```
Comment (allanhaywood) [inline PlotTalk/Src/accountmanager.h:26]: Same comment about pass by value. I'll stop commenting going forward, you can apply the same feedback elsewhere where it applies.

File: PlotTalk/Src/createacctpage.cpp [hunk 13]
```diff
@@ -22,32 +25,26 @@ CreateAcctPage::~CreateAcctPage()
...
-
+       AccountManager *check= AccountManager::getInstance();
        if(ui->PasswordBox1->text()==ui->PasswordBox2->text())//must also check to see if password matches requirements
        {
-           password=ui->PasswordBox1->text();
-           AccountManager *addNew= AccountManager::getInstance();
-           if(addNew->createAccount(firstName,lastName,email,handle,password))
+           switch(check->checkFields(firstName, lastName, handle, email, password))
+           {
+           case 1:
```
Comment (allanhaywood) [inline PlotTalk/Src/createacctpage.cpp:51]: What is case 1, case 2, case 3?

At the very least add comments to what these mean, even better use an ENUM instead, and the code will be self documenting.

http://stackoverflow.com/questions/3019153/how-do-i-use-an-enum-value-on-a-switch-statement-in-c

**Comment by allanhaywood** [comment]
Other than the specific inline comments, here is some general feedback.
Ideally use doxygen format comments.
Ideally use singleton.h to make a class a singleton (I know it came after you had implemented singletons, but I believe it is easier to use, and we should consolidate on one method). First ask if the class should be a singleton.
Ideally store hashes instead of plaintext passwords: http://doc.qt.io/qt-5/qcryptographichash.html

Avoid using pointers when it is not necessary (Exception for when interacting with QT and following their best practices for that, which does involve pointers).
Prefer using references instead of pointers, when that behavior is needed.
Avoid newing up objects if they are fairly small objects that will fit on the stack.

**Comment by reave006** [comment]
Merge conflicts with master have been resolved.

**Review by allanhaywood — CHANGES_REQUESTED** [review]
Sorry, didn't realize I had to submit reviews again.

**Review by allanhaywood — APPROVED** [review]
